packages/llmd-serving/src/deployments/model.ts

```diff
@@ -85,24 +102,163 @@ const structuredCloneWithMainContainer = (
 
 export const applyModelArgs = (
   llmdInferenceService: LLMInferenceServiceKind,
-  modelArgs?: string[],
+  modelArgs?: RuntimeArgsFieldData,
 ): LLMInferenceServiceKind => {
-  if (!modelArgs) {
-    return llmdInferenceService;
-  }
   const { result, mainContainer } = structuredCloneWithMainContainer(llmdInferenceService);
-  mainContainer.args = modelArgs;
+  if (!modelArgs?.enabled) {
+    delete mainContainer.args;
+    return result;
+  }
+  mainContainer.args = modelArgs.args;
   return result;
 };
 
 export const applyModelEnvVars = (
   llmdInferenceService: LLMInferenceServiceKind,
-  modelEnvVars?: { name: string; value: string }[],
+  modelEnvVars?: EnvironmentVariablesFieldData,
 ): LLMInferenceServiceKind => {
-  if (!modelEnvVars) {
-    return llmdInferenceService;
-  }
   const { result, mainContainer } = structuredCloneWithMainContainer(llmdInferenceService);
-  mainContainer.env = modelEnvVars;
+  if (!modelEnvVars?.enabled) {
+    delete mainContainer.env;
+    return result;
+  }
+  mainContainer.env = modelEnvVars.variables;
+  return result;
+};
+
+export const getModelLocationUri = (deployment: LLMInferenceServiceKind): string | undefined => {
+  return deployment.spec.model.uri;
+};
+
+const extractAdditionalFields = (deployment: LLMInferenceServiceKind): Record<string, string> => {
+  const additionalFields: Record<string, string> = {};
+  const { model } = deployment.spec;
+
+  const connectionType = getConnectionTypeFromUri(model.uri);
+  if (connectionType === ModelServingCompatibleTypes.S3ObjectStorage) {
+    additionalFields.modelPath =
+      deployment.metadata.annotations?.['opendatahub.io/connection-path'] || '';
+  }
+
+  if (connectionType === ModelServingCompatibleTypes.OCI) {
+    additionalFields.modelUri = model.uri || '';
+  }
+
+  return additionalFields;
+};
+
+export const extractModelLocationData = (deployment: {
+  model: LLMInferenceServiceKind;
+}): ModelLocationData => {
+  const uri = getModelLocationUri(deployment.model);
+  if (uri && isPVCUri(uri)) {
+    return {
+      type: ModelLocationType.PVC,
+      fieldValues: { URI: uri },
+      additionalFields: {
+        pvcConnection: getPVCNameFromURI(uri),
+      },
+    };
+  }
+  const imagePullSecrets = deployment.model.spec.template?.containers?.find(
+    (container) => container.name === 'imagePullSecrets',
+  );
```

Comment:
Yeah the `imagePullSecret` should be in `spec.template?.imagePullSecrets[x]`. This needs to be fixed or else we'll never get the imagePullSecret for OCI?

```suggestion
// get the first one i guess? that's what we've been doing, technically could be wrong
  const imagePullSecrets = 
    deployment.model.spec.template?.imagePullSecrets?.length > 0 
      ? deployment.model.spec.template?.imagePullSecrets[0] 
      : undefined
```

<img width="653" height="760" alt="Image" src="https://github.com/user-attachments/assets/48e78999-b11d-4339-b452-7eda9a823fa0" />